File: mvc-sample/src/main/kotlin/suzp1984/github/io/jbig_kotlin/states/JbigDbState.java
package suzp1984.github.io.jbig_kotlin.states;

import java.util.Collection;
import java.util.List;

/**
 * Created by suzhenxi on 11/28/2016.
 */

public interface JbigDbState extends BaseState {
    List<byte[]> getJbigDbs();
    byte[] getJbigAtPosition(int position);

    void putJbig(byte[] jbig);
    void putJbigs(Collection<byte[]> jbigs);
    void deleteJbig(byte[] jbig);

    public static class JbigDbAddEvent {
    }

    public static class JbigDbDeleteEvent {

    }
}

File: mvc-sample/src/main/kotlin/suzp1984/github/io/jbig_kotlin/states/JbigDbState.kt
package suzp1984.github.io.jbig_kotlin.states

/**
 * Created by suzhenxi on 11/28/2016.
 */

interface JbigDbState : BaseState {
    val jbigDbs: List<ByteArray>
    fun getJbigAtPosition(position: Int): ByteArray

    fun putJbig(jbig: ByteArray)
    fun putJbigs(jbigs: Collection<ByteArray>)
    fun deleteJbig(jbig: ByteArray)

    class JbigDbAddEvent

    class JbigDbDeleteEvent
}
